tests/gradients/finite_diff/test_finite_difference_shot_vec.py

```diff
@@ -396,7 +396,7 @@ def cost6(x):
         circuits = [qml.QNode(cost, dev) for cost in (cost1, cost2, cost3, cost4, cost5, cost6)]
 
         transform = [qml.math.shape(qml.gradients.finite_diff(c, h=h_val)(x)) for c in circuits]
-        expected = [(3, 3), (1, 3, 3), (3, 2, 3), (3, 3, 4), (1, 3, 3, 4), (3, 2, 3, 4)]
+        expected = [(3, 3), (3, 1, 3), (3, 2, 3), (3, 3, 4), (3, 1, 3, 4), (3, 2, 3, 4)]
```

Comment:
Here it actually looks like the original tests were expecting incorrect output shapes. According to https://github.com/PennyLaneAI/pennylane/pull/5418, the level of priority for dimensions from outer dimension to inner dimension is:
```
1. Quantum Script in batch. No squeezing.
2. Shot choice in a shot vector. Squeezed if no shot vector.
3. Measurement in the quantum script. Squeezed out if only one measurement.
4. Parameter broadcasting. Squeezed out if no parameter-broadcasting.  Adds to array shape instead of adding tuple nesting.
5. Fundamental measurement shape.
```
Here for circuits `cost1` and `cost5`, the batch dimension 3 should be outside the dimension that corresponds to the number of measurements. For some reason, if the measurements are returned as a list, the dimension is not expected to be squeezed out even if the list contains only one item. Does anyone understand why?

Reply:
Do you mean that I should create a separate issue and link it to this PR?

Reply:
As far as I understand, the code _actually_ produces what's called "expected" here, right?
So why does changing the "expected" not make the test fail. And if the "expected" as it currently is matches code behaviour but not the docs, we should fix that independently (but in particular we should fix it :sweat_smile: )

Reply:
Sounds like we need to discuss what we consider to be correct behaviour before any changes are merged.

> iirc, this is because we assume that if a user says [qml.probs([0, 1])] they really want a list-valued output for some reason.
If this is the case, then we need to update the documentation to reflect this.

Reply:
> As far as I understand, the code _actually_ produces what's called "expected" here, right? So why does changing the "expected" not make the test fail. And if the "expected" as it currently is matches code behaviour but not the docs, we should fix that independently (but in particular we should fix it 😅 )

Without my change, the code currently produces incorrect results, and the test is passing because it is expecting incorrect results. After my change, the tests failed, because now the code actually produces correct results, but the tests are expecting the wrong things. That's why I updated the tests.

Reply:
Consider this circuit:
```
dev = qml.device("default.qubit", shots=[10, 10, 10, 10])

@qml.qnode(dev)
def circuit(inputs):
    qml.RX(inputs, wires=0)
    qml.RY(inputs, wires=1)
    qml.CNOT(wires=[0, 1])
    return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))]

x = np.zeros(3)
res = circuit(x)
```
If you examine the output shape, you see
```
>>> qml.math.shape(res)
(4, 2, 3)
```
Where the three dimensions correspond to shot vector length, number of measurements, parameter broadcasting, which follows the order of priority outlined in https://github.com/PennyLaneAI/pennylane/pull/5418. However, if you reduce the number of measurements to 1:
```
@qml.qnode(dev)
def circuit(inputs):
    qml.RX(inputs, wires=0)
    qml.RY(inputs, wires=1)
    qml.CNOT(wires=[0, 1])
    return [qml.expval(qml.PauliZ(0))]

x = np.zeros(3)
res = circuit(x)
```
This happens:
```
>>> qml.math.shape(res)
(1, 4, 3)
```
See that the dimension for the number of measurements is placed outside the dimension of the shot vector, which violates the return type specifications, and also inconsistent.